fs-tep-api/src/main/java/com/cgi/eoss/fstep/api/controllers/UserPreferencesApi.java

```diff
@@ -33,10 +33,10 @@
     @Override
     @PreAuthorize("hasAnyRole('CONTENT_AUTHORITY', 'ADMIN') or hasPermission(#userPreference, 'administration')")
     void delete(@P("userPreference") UserPreference userPreference);
-
+   
     @Override
-    @Query("select t from UserPreference t where t.owner=user")
-    Page<UserPreference> findByOwner(@Param("owner") User user, Pageable pageable);
+    @Query("select t from UserPreference t where t.owner=user and t.name=name and t.type=type")
+    Page<UserPreference> findByOwner(@Param("owner") User user, @Param("name")String name, @Param("type")String type, Pageable pageable);
```

Comment:
I think this could be named a more generic `search` or something, since you added multiple non-owner search fields. You might have to check the generated API endpoint name though.

If you want to keep the `findBy` Spring Data naming convention it should probably be kept in sync with the `@Query` - `findByOwnerAndNameAndType`.

Reply:
Agreed, this seems out of sync with the semantics we are trying to convey. Renaming it to search.